lesson11/index.js

```diff
@@ -0,0 +1,84 @@
+const ul = document.getElementById('js-list');
+
+const createLoading = () => {
+    const li = document.createElement("li");
+    const loadingImg = document.createElement('img');
+
+    li.style.listStyle = "none";
+    li.id = "js-loading";
+    loadingImg.src = 'loading-circle.gif';
+    ul.appendChild(li).appendChild(loadingImg);
+};
+
+const removeLoading = () => document.getElementById('js-loading').remove();
+
+const createErrorMessage = (error) => {
+    const li = document.createElement("li");
+    li.textContent =`エラー内容：${error.message}`;
+    ul.appendChild(li);
+    console.error(error.message);
+};
+
+/**
+ * Async関数 
+ * @return {Array} 
+ */
+const jsonUrl = 'https://myjson.dit.upm.es/api/bins/4tpn';
+const fetchData = async () => {
+    try {
+        const response = await fetch(jsonUrl) // 中身はresponseオブジェクトを含むpromise
+        const json = await response.json(); //response オブジェクトから JSON を抽出
+        return json.data;
+
+    } catch(error) {
+        console.error(error);
+    } 
```

Comment:
```suggestion
    } catch(error) {
        throw new Error(error);
    } 
```
IMO
I thought it would be better to throw an error here.

The reason for this is that if an error occurs during the execution of fetchData
In the current code, the variable "data" in fetchListData is undefined.
The error "Cannot read properties of undefined (reading 'length')" occurs, and the code transitions to catch.

However, by throwing an error, the if statement is not executed and the process can be passed directly to catch.

Reply:
Fixed the error part.
[8cd7250](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/11/commits/8cd7250bed03d4eaf7b05db8ef558ad9b9e0ecaf)